### PR TITLE
[react-native] Update EventEmitter for RN 0.64

### PR DESCRIPTION
**Pull request by newyankeecodeshop**
A few major changes were made to the `EventEmitter` interface in RN 0.64:
https://github.com/facebook/react-native/commit/87a2e29f5928c2e09ac9a98c54732d5f697d8e61#diff-a121d2bb39bffcd70cfa7d43313e06f241d83b27acba0b551dcca9b19f605d6c
https://github.com/facebook/react-native/commit/b11d6ecbb8bb2f0d6f423be6775e587f4e9b1c4d#diff-a121d2bb39bffcd70cfa7d43313e06f241d83b27acba0b551dcca9b19f605d6c

This PR updates the types to reflect those changes.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/facebook/react-native/blob/0.64-stable/Libraries/vendor/emitter/EventEmitter.js
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header. (Already at 0.64)

### CODE REVIEW

**Comment by kuasha420** [comment]
@newyankeecodeshop Looks good to me according to the upstream source. Would be nice if you added test.

File: types/react-native/test/index.tsx [hunk 1]
```diff
@@ -918,6 +918,8 @@ deviceEventEmitterStatic.addListener('keyboardWillShow', data => true, {});
 const androidEventEmitter = new NativeEventEmitter();
 const sub1 = androidEventEmitter.addListener('event', (event: object) => event);
 const sub2 = androidEventEmitter.addListener('event', (event: object) => event, {});
+androidEventEmitter.listenerCount('event');
```
Comment (ca057) [inline types/react-native/test/index.tsx:921]:
```suggestion
androidEventEmitter.listenerCount('event'); // $ExpectType number
```

**Review by ca057 — APPROVED** [review]
I would make the return type explicit in the tests, otherwise it looks good to me!

**Review by ca057 — APPROVED** [review]

**Comment by newyankeecodeshop** [comment]
Ready to merge

**Pull request merged**